matRad_shift.m

```diff
@@ -0,0 +1,81 @@
+function [projCoord,idx,targetPoint, sourcePoint] = matRad_shift(initIx,dim,sourcePoint_bev,targetPoint_bev,isoCenter, res, Dx, Dz, rotMat)
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% This function projects a point on a certain ray and returns both the index
+% of the projected point in the reference system of the ct cube and its
+% coordinates
+%
+% call
+%   [finalWeight, X1, sigma_sub, radius, posx, posy, numOfSub] = 
+%                                  matRad_calcWeights(sigma_ray, n, method)
+%
+% input
+%   initIx:             initial indices of the points
+%   cubeDim:            dimension of the ct cube (i.e. ct.cubeDim)
+%   sourcePoint_bev:    source point of the ray in bev
+%   targetPoint_bev:    target point of the ray in bev
+%   isoCenter:          isocenter coordinates
+%   res:                resolution in x,y, and z direction
+%   Dx:                 displacement on x axis
+%   Dz:                 displacement on z axis
+%
+% output
+%   idx:        projected indeces
+%   projCoord:  projected coordinates
+%
+% References
+%
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Copyright 2017 the matRad development team.
+%
+% This file is not part of the offical matRad release
+%
+% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+% add offset to target and source point in bev
+
+Dx = -Dx;
+Dz = -Dz;
+
+targetPoint_bevVec = targetPoint_bev(2).*ones([length(Dx) 3]);
+targetPoint_bevVec(:,1) = bsxfun(@plus,Dx,targetPoint_bev(1));
+targetPoint_bevVec(:,3) = bsxfun(@plus,Dz',targetPoint_bev(3));
```

Comment:
I am a little confused... why is it not 

targetPoint_bevVec(:,2) = targetPoint_bev(2).*ones([length(Dx) 3]);

I think that line 44 directly overwrites what you have written in line 43 making targetPoint_bevVec(:,2) all zeros.